components/hooks/useRenderQuestionField.tsx

```diff
@@ -0,0 +1,391 @@
+        return (
+          <NumberRangeQuestionComponent
+            parsedQuestion={parsed}
+            numberRange={numberRangeProps?.numberRange}
+            handleNumberChange={numberRangeProps?.handleNumberChange}
+            startPlaceholder="start"
+            endPlaceholder="end"
+          />
+        );
+      }
+      break;
+
+    case CURRENCY_QUESTION_TYPE:
+      if (parsed.type === 'currency' && currencyProps) {
+        return (
+          <CurrencyQuestionComponent
+            parsedQuestion={parsed}
+            inputCurrencyValue={currencyProps?.inputCurrencyValue}
+            setInputCurrencyValue={currencyProps?.setInputCurrencyValue}
+          />
+        );
+      }
+      break;
+
+    case URL_QUESTION_TYPE:
+      if (parsed.type === 'url' && urlProps) {
+        const { minLength, maxLength, pattern } = parsed.attributes || {};
+        return (
+          <FormInput
+            name="urlInput"
+            type="url"
+            label="url"
+            placeholder="url"
+            value={urlProps?.urlValue ?? ''
+            }
+            onChange={urlProps?.handleUrlChange}
+            minLength={minLength}
+            maxLength={maxLength}
+            pattern={pattern}
+          />
+        );
+      }
+      break;
+
+    case EMAIL_QUESTION_TYPE:
+      if (parsed.type === 'email' && emailProps) {
+        const { minLength, maxLength, pattern } = parsed.attributes || {};
+        return (
+          <FormInput
+            name="emailInput"
+            type="email"
+            label="email"
+            placeholder="email"
+            value={emailProps?.emailValue ?? ''
+            }
+            onChange={emailProps?.handleEmailChange}
+            minLength={minLength}
+            maxLength={maxLength}
+            pattern={pattern}
+          />
+        );
+      }
+      break;
+
+    case BOOLEAN_QUESTION_TYPE:
+      if (parsed.type === 'boolean' && booleanProps) {
+        return (
+          <BooleanQuestionComponent
+            parsedQuestion={parsed}
+            selectedValue={booleanProps?.yesNoValue}
+            handleRadioChange={booleanProps?.handleBooleanChange}
+          />
+        );
+      }
+      break;
+
+    case TYPEAHEAD_QUESTION_TYPE:
+      if (parsed.type === 'typeaheadSearch' && typeaheadSearchProps) {
+        return (
+          <AffiliationSearchQuestionComponent
+            parsedQuestion={parsed}
+            affiliationData={typeaheadSearchProps.affiliationData}
+            otherAffiliationName={typeaheadSearchProps.otherAffiliationName}
+            otherField={typeaheadSearchProps.otherField}
+            setOtherField={typeaheadSearchProps.setOtherField}
+            handleAffiliationChange={typeaheadSearchProps.handleAffiliationChange}
+            handleOtherAffiliationChange={typeaheadSearchProps.handleOtherAffiliationChange}
+          />
+        );
+      }
+      break;
+
+    default:
+      return <p>Unsupported question type </p>;
```

Comment:
Should we maybe log this elsewhere? If we never matched a question type, this would mean that this is most likely an error or bug somewhere? If we log it, it's more likely that we can catch and deal with it.

We can for example log the invalid question type that could not be matched.